examples/codegen/elixir_api/lib/elixir_api_web/controllers/blocked_times_export_controller.ex

```diff
@@ -0,0 +1,69 @@
+defmodule ElixirApiWeb.BlockedTimesExportController do
+  @moduledoc false
+
+  use ElixirApiWeb, :controller
+
+  action_fallback(ElixirApiWeb.FallbackController)
+
+  # add aliases here
+
+  def create(conn, _params) do
```

Comment:
This is a great start!

One obvious issue I'm seeing is that after the developer implements the missing parts they can't regenerate the thing without undoing their work. 

The answer to this is ofc better modularity

We should make the thing modular enough that The controller code is actually dumb enough to not need any manual changes ever. That would require us to agree on some conventions as to what the code in the controller action should look like and what should it call into and what the error handling should look like.

same thing is obiously true about the views, the tests, the router, etc.
There are many ways to achieve this separation between generated and nongenerated code.
It's not ease to think of optimal patterns for this from the top of my head.

Another thing that comes to my mind when trying to think of the patterns is that
if we make the generator smart enough to know that If an action/test/view already exists, then It should not regenerate it then it would be much easier to make the separation natural and not requiring splitting things into many files. Wdyt? I know it's not easy to make a foolproof mechanism like that without actually parsing the code (or using mysterious nonstandard extended regexes), so it might not be worth to introduce this complexity.

Reply:
If we want to get the most value from the spec->code workflow then we have to enable painfree, nondestructive generation. I'm stating the obvious here - generation of a new endpoint should not destroy a whole controller, etc